src/contexts/RDKitContext.tsx

```diff
@@ -22,18 +22,27 @@ export const RDKitProvider: React.FC<RDKitProviderProps> = ({ initialRdkitInstan
 
   useEffect(() => {
     let isProviderMounted = true;
-    if (!initialRdkitInstance && globalThis.initRDKitModule) {
-      globalThis.initRDKitModule().then((loadedRDKit) => {
-        if (isProviderMounted) {
-          setRDKit(loadedRDKit);
-          globalThis.rdkitProviderGlobals = {
-            jsMolCacheEnabled: !!enableJsMolCaching,
-            jsMolCache: enableJsMolCaching ? {} : null,
-            maxJsMolsCached: maxJsMolsCached ?? MAX_CACHED_JSMOLS,
-          };
-        }
-      });
-    }
+
+    const initialise = async () => {
+      let loadedRDKit: RDKitModule | undefined;
+
+      if (!initialRdkitInstance && globalThis.initRDKitModule) {
+        loadedRDKit = await globalThis.initRDKitModule();
```

Comment:
@RamziWeslati Do we need to check `isProviderMounted` before calling `initRDKitModule` ?

Reply:
No need, it's there to avoid setting state on unmounted components